src/python/pants/backend/python/goals/export.py

```diff
@@ -78,9 +77,69 @@ def debug_hint(self) -> str | None:
         return self.resolve_name
 
 
+@rule_helper
+async def _do_export(
+    requirements_pex: Pex,
+    pex_pex: PexPEX,
+    dest: str,
+    resolve_name: str,
+    qualify_path_with_python_version: bool,
+) -> ExportResult:
+    # Get the path to the interpreter, and the full python version (including patch #).
+    res = await Get(
+        ProcessResult,
+        PexProcess(
+            pex=requirements_pex,
+            description="Get interpreter path and version",
+            argv=[
+                "-c",
+                "import sys; print(sys.executable); print('.'.join(str(x) for x in sys.version_info[0:3]))",
```

Comment:
This seemed like a more robust way to get a kosher interpreter then the previous line `interpreter = cast(PythonExecutable, requirements_pex.python)` and its clarifying comment below.

Reply:
I'm not so sure about that. The internal-onlyness is a critical thing here that is just as obscure still.

Reply:
The internal-onlyness was critical in the previous method of using `requirements_pex.python`. Why is it critical here? This must give us *some* compatible interpreter in any case, surely?

Reply:
I'm trying to point out you replaced a comment and re-use of an already computed and correct interpreter with another comment elsewhere about internal onlyness _and_ a re-compute.

As far as criticality goes, it's critical always and until creating and storing a fully zipped PEX is fast and efficient. It never has been either. Zipping is always slow, and packed is needed for remote caching to work ~at all / not be storing both a wheel zip and a copy of that zipped up in 18 different non-packed PEX zips in LMDB.

Reply:
Ah, fair enough. In this case I interpreted "critical" to mean "for correctness" - i.e., `requirements_pex.python` is only guaranteed to be populated because of internal-onlyness. This code will blow up otherwise. Getting the path to an interpreter by running the pex instead, as this now does, sidesteps the need for this assumption and won't blow up if the Pex is not internal-only.

You're referring to internal-onlyness being critical for performance, which it definitely is.

But since performance is important, it may be better to blow up, as it indicates a code bug. I'll change this.